### PR TITLE
INTYGFV-15820: Updated how rekostatus is retrieved for patient view

### PR DESCRIPTION
**Pull request by mhornfeldt**

### CODE REVIEW

File: apps/rehabstod/src/store/sickLeaveApi.ts [hunk 1]
```diff
@@ -109,6 +110,14 @@ const sickLeaveApi = api.injectEndpoints({
         }
       },
     }),
+    getRekoStatusForPatient: builder.query<RekoStatus, { patientId: string; endDate: string; startDate: string }>({
+      keepUnusedDataFor: 0,
```
Comment (SamuelTilly) [inline apps/rehabstod/src/store/sickLeaveApi.ts:114]: Perhaps we could use the cache instead? We could add a tag with specific identifier that gets invalidated with setRekoStatus.

setRekoStatus:
```typescript
invalidatesTags: (result, error, {patientId}) => [{type: 'RekoStatus', id: patientId}],
```

getRekoStatusForPatient:
```typescript
providesTags: (result, error, {patientId}) => [{ type: 'RekoStatus', id: patientId }],
```

File: apps/rehabstod/src/pages/Patient/components/SelectRekoStatus/SelectRekoStatus.tsx [hunk 2]
```diff
@@ -1,42 +1,44 @@
...
       setRekoStatus({ patientId, status: type, sickLeaveTimestamp, filter })
       updateSavedRekoStatus(type.id)
-      dispatch(updateRekoStatusId(type.id))
     }
   }
 
+  useEffect(() => {
+    updateSavedRekoStatus(rekoStatusFromDatabase?.status.id)
+  }, [rekoStatusFromDatabase])
+
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/SelectRekoStatus/SelectRekoStatus.tsx:41]: Not sure if we need to store `rekostatus` in a local state. We could use the state that the API provides.

```typescript
const savedRekoStatus = rekoStatusFromDatabase?.status.id || emptyRekoStatus?.id
```

File: apps/rehabstod/src/pages/Patient/components/SelectRekoStatus/SelectRekoStatus.tsx [hunk 3]
```diff
@@ -1,42 +1,44 @@
...
   patientId: string
   endDate: string
+  startDate: string
   rekoStatusTypes: RekoStatusType[]
   description: string
   disabled?: boolean
 }) {
   const [setRekoStatus] = useSetRekoStatusMutation()
-  const { rekoStatusId } = useAppSelector((rootState) => rootState.sickLeave)
+  const { data: rekoStatusFromDatabase } = useGetRekoStatusForPatientQuery({ patientId, endDate, startDate })
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/SelectRekoStatus/SelectRekoStatus.tsx:24]: "FromDatabase" is a bit verbose, I guess everything from the API is stored in a database.

File: apps/rehabstod/src/pages/Patient/components/PatientRekoStatus.tsx [hunk 4]
```diff
@@ -16,8 +12,6 @@ export function PatientRekoStatus({
   isDoctor: boolean
 }) {
   const { data: populatedFilters } = useGetSickLeavesFiltersQuery()
-  const { state } = useLocation()
```
Comment (SamuelTilly) [inline apps/rehabstod/src/pages/Patient/components/PatientRekoStatus.tsx:19]: This state can be removed from `apps\rehabstod\src\pages\CurrentSickLeaves\components\TableBodyRows.tsx`as well.

**Review by SamuelTilly — CHANGES_REQUESTED** [review]
Awesome that this could be fixed! found some minor things that could be changed

File: apps/rehabstod/src/store/sickLeaveApi.ts (hunk 1 above)
Comment (mhornfeldt) [reply apps/rehabstod/src/store/sickLeaveApi.ts:114]: I remember Matias mentioned at a meeting that we shouldn't use the cache at all for the patientview.

Comment (SamuelTilly) [reply apps/rehabstod/src/store/sickLeaveApi.ts:114]: Alright, let's keep the `keepUnusedDataFor`. I think we still need to invalidate `getRekoStatusForPatient` when `setRekoStatus` gets called.

Comment (mhornfeldt) [reply apps/rehabstod/src/store/sickLeaveApi.ts:115]: This gives a ts error

Comment (SamuelTilly) [reply apps/rehabstod/src/store/sickLeaveApi.ts:115]: Probably missing 'RekoStatus' in the list of tags in `api.ts` ?

**Review by SamuelTilly — APPROVED** [review]

**Pull request merged**